lib/arch/generic/dec_loop.c

```diff
@@ -0,0 +1,46 @@
+// Read source 4 bytes at a time
+// Since we might be writing one byte more than needed,
+// we need to make sure there will still be some room
+// for one extra byte in o.
+// This will be the case if srclen > 0 when the loop
+// is exited
```

Comment:
This is true given the contract in the API. The output buffer size must be at least 3/4 the size of the input buffer.